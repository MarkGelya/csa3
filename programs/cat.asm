        JMP     start
buff:   BYTE    0xFF
start:
loop:
        IN      1
        OUT     1
        JEOF    end
        JMP     loop
end:
        HLT
