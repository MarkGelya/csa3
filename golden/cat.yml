in_source: |
  ; cat
          JMP     start
  start:
  loop:
          IN      1
          JEOF    end
          OUT     1
          JMP     loop
  end:
          HLT


in_stdin: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras non elit iaculis, fringilla massa in, lacinia tellus. Praesent eu enim in leo hendrerit faucibus id in felis. In venenatis dui at sollicitudin euismod.'
out_log: ''
out_stdout: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras non elit iaculis, fringilla massa in, lacinia tellus. Praesent eu enim in leo hendrerit faucibus id in felis. In venenatis dui at sollicitudin euismod.'
